src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -1540,8 +1539,10 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
     }
 
     void visit_DictItem(const ASR::DictItem_t& x) {
-        ASR::Dict_t* dict_type = ASR::down_cast<ASR::Dict_t>(
-                                    ASRUtils::expr_type(x.m_a));
+        ASR::Dict_t *dict_type = ASR::is_a<ASR::Const_t>(*ASRUtils::expr_type(x.m_a))
+                                     ? ASR::down_cast<ASR::Dict_t>(ASRUtils::type_get_past_const(ASRUtils::expr_type(x.m_a)))
```

Comment:
`type_get_past_const()` already has a check `ASR::is_a<ASR::Const_t>()`, so the check for `ASR::is_a<ASR::Const_t>` here is redundant and thus not needed.

Reply:
Similarly update others.

Reply:
I did not get what you meant @Shaikh-Ubaid. Can you please show the changes? How will I know if the dictionary is not `Const` if I don't check the type? Please guide.

Reply:
I think, @Shaikh-Ubaid, suggested to just do:
```c++
ASR::Dict_t *dict_type = ASR::down_cast<ASR::Dict_t>(ASRUtils::type_get_past_const(ASRUtils::expr_type(x.m_a)));
```

Reply:
So `ASRUtils::type_get_pas_const()` returns the same passed value if the value is not `Const`. Got it @Thirumalai-Shaktivel! :+1:

Reply:
Yup!